runtime error: when "SetGlobalFormatter" followed by  "SetLogger"

1. What version of Go and beego are you using (`bee version`)?
beego version is v2.0.7
```
$ bee version
______
| ___ \
| |_/ /  ___   ___
| ___ \ / _ \ / _ \
| |_/ /|  __/|  __/
\____/  \___| \___| v2.0.4

├── GoVersion : go1.18.4
├── GOOS      : linux
├── GOARCH    : amd64
├── NumCPU    : 4
├── GOPATH    : 
├── GOROOT    : /usr/local/go
├── Compiler  : gc
└── Date      : Monday, 19 Dec 2022
```

2. What operating system and processor architecture are you using (`go env`)?

```
$ uname -a 
Linux  5.8.0-41-generic #46~20.04.1-Ubuntu SMP Mon Jan 18 17:52:23 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

3. What did you do?

run cmd `bee api demo ; cd demo ` and update beego version to 2.0.7

edit main.go :
``` golang
package main

import (
	_ "demo/routers"

	"github.com/beego/beego/v2/core/logs"
)

func main() {
	f := &logs.PatternLogFormatter{
		Pattern:    "%w[%T]:%m",
		WhenFormat: "2006/01/02 15:04:05.123",
	}
	logs.RegisterFormatter("pattern", f)
	logs.SetGlobalFormatter("pattern")
	logs.SetLogger(logs.AdapterMultiFile, `{"filename":"test.log"}`)
	logs.Info("hi")
}
```
run `go mod tidy ; go run main.go`

4. What did you expect to see?

log in test.log by "%w[%T]:%m"

5. What did you see instead?

```linux
$ go run main.go 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x110 pc=0x6b3fe0]

goroutine 1 [running]:
github.com/beego/beego/v2/core/logs.(*fileLogWriter).SetFormatter(...)
	/home/fuchangjie/go/pkg/mod/github.com/beego/beego/v2@v2.0.7/core/logs/file.go:108
github.com/beego/beego/v2/core/logs.(*multiFileLogWriter).SetFormatter(0x90d780?, {0xa6e3a0?, 0xc000037c60?})
	/home/fuchangjie/go/pkg/mod/github.com/beego/beego/v2@v2.0.7/core/logs/multifile.go:91 +0x20
github.com/beego/beego/v2/core/logs.(*BeeLogger).setLogger(0xc00015c000, {0x9a0325, 0x9}, {0xc00012ff60?, 0x1?, 0xde0000c000037c60?})
	/home/fuchangjie/go/pkg/mod/github.com/beego/beego/v2@v2.0.7/core/logs/log.go:202 +0x262
github.com/beego/beego/v2/core/logs.(*BeeLogger).SetLogger(0x90d780?, {0x9a0325?, 0x99e9c0?}, {0xc00012ff60?, 0x401230?, 0x0?})
	/home/fuchangjie/go/pkg/mod/github.com/beego/beego/v2@v2.0.7/core/logs/log.go:223 +0x12d
github.com/beego/beego/v2/core/logs.SetLogger(...)
	/home/fuchangjie/go/pkg/mod/github.com/beego/beego/v2@v2.0.7/core/logs/log.go:703
main.main()
	/home/fuchangjie/git/beego/demo/main.go:16 +0x112
exit status 2

```
Is "SetGlobalFormatter" and "SetLogger" cannot exist at the same time?

anybody can help ? thanks a  lot

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,8 @@
 # developing
 - [Fix 5117: support write though cache](https://github.com/beego/beego/pull/5117)
+- [add read through for cache module](https://github.com/beego/beego/pull/5116)
+- [add singleflight cache for cache module](https://github.com/beego/beego/pull/5119)
+- [Fix 5129: must set formatter after init the logger](https://github.com/beego/beego/pull/5130)
 
 # v2.0.7
 - [Upgrade github.com/go-kit/kit, CVE-2022-24450](https://github.com/beego/beego/pull/5121)
@@ -16,9 +19,6 @@ Note: now we force the web admin service serving HTTP only.
 - [Fix 5012: fix some bug, pass []any as any in variadic function](https://github.com/beego/beego/pull/5012)
 - [Fix 5022: Miss assigning listener to graceful Server](https://github.com/beego/beego/pull/5028)
 - [Fix 4955: Make commands and Docker compose for ORM unit tests](https://github.com/beego/beego/pull/5031)
-- [add read through for cache module](https://github.com/beego/beego/pull/5116)
-- [add singleflight cache for cache module](https://github.com/beego/beego/pull/5119)
-
 
 # v2.0.4
 

--- a/core/logs/log.go
+++ b/core/logs/log.go
@@ -41,8 +41,6 @@ import (
 	"strings"
 	"sync"
 	"time"
-
-	"github.com/pkg/errors"
 )
 
 // RFC5424 log message levels.
@@ -193,20 +191,20 @@ func (bl *BeeLogger) setLogger(adapterName string, configs ...string) error {
 
 	lg := logAdapter()
 
+	err := lg.Init(config)
+	if err != nil {
+		return err
+	}
+
 	// Global formatter overrides the default set formatter
 	if len(bl.globalFormatter) > 0 {
 		fmtr, ok := GetFormatter(bl.globalFormatter)
 		if !ok {
-			return errors.New(fmt.Sprintf("the formatter with name: %s not found", bl.globalFormatter))
+			return fmt.Errorf("the formatter with name: %s not found", bl.globalFormatter)
 		}
 		lg.SetFormatter(fmtr)
 	}
 
-	err := lg.Init(config)
-	if err != nil {
-		fmt.Fprintln(os.Stderr, "logs.BeeLogger.SetLogger: "+err.Error())
-		return err
-	}
 	bl.outputs = append(bl.outputs, &nameLogger{name: adapterName, Logger: lg})
 	return nil
 }

